.circleci/config.yml

```diff
@@ -13,6 +13,14 @@ aliases:
 
 # Reuseable commands
 commands:
+
+  check-changed-files-or-halt:
+    parameters:
+      pattern:
+        type: string
+    steps:
+      - run: git show -m HEAD --name-only --pretty="" | egrep -q -v '<< parameters.pattern >>' || circleci step halt
```

Comment:
Will this cancel the run if there is no file changed with the content `<< parameters.pattern >>` ? If so, I don't think this is a correct condition: that'll be the case in most situations.

Maybe a better condition is: filter out all files in `.github` and check if there are any files left which have changed?

Reply:
- git show list all files involved in the last PR
- grep -q -v set its exit status to success if there are no files matching `<< parameters.pattern >>` and the step will halt

Reply:
Right. And the pattern you're using is `^.github/`; so, if no files match that, the step will halt. But when I only change one of the workflow scripts (e.g. `lib/LedgerSMB/Scripts/setup.pm`) then the build will be cancelled for not matching `.github/`, right?

Reply:
The build is cancelled only if .github is the only thing in the modified files.

Reply:
ah! sorry about misunderstanding your code! I see I overlooked the `-v` switch's effects.